download.sh

```diff
@@ -0,0 +1,106 @@
+#! /usr/bin/env bash
+USAGE="Usage: ./download.sh <DUMP_DIR>
+
+Download the latest Wikipedia Enterprise HTML dumps.
+
+Arguments:
+    <DUMP_DIR>  An existing directory to store dumps in. Dumps will be grouped
+                into subdirectories by date, and a link 'latest' will point to
+                the latest complete dump subdirectory, if it exists.
+
+Environment Variables:
+    LANGUAGES   A space-separated list of wikipedia language codes to download
+                dumps of.
+                Defaults to the languages in 'article_processing_config.json'.
+                See <https://meta.wikimedia.org/wiki/List_of_Wikipedias>.
+
+Exit codes:
+    0   The lastest dumps are already present or were downloaded successfully.
+    1   Argument error.
+    16  Some of languages were not available to download. The latest dump may
+        be in progress, or some of the specified languages may not exist.
+    _   Subprocess error.
+"
+
+set -euo pipefail
+# set -x
+
+if [ -z "${1:-}" ]; then
+    echo -n "$USAGE" >&2
+    exit 1
+fi
+
+# The parent directory to store groups of dumps in.
+DUMP_DIR=$(readlink -f "$1")
+shift
+
+if [ ! -d "$DUMP_DIR" ]; then
+    echo "DUMP_DIR '$DUMP_DIR' does not exist" >&2
+    exit 1
+fi
+
+# Ensure we're running in the directory of this script.
+SCRIPT_PATH=$(dirname "$0")
+cd "$SCRIPT_PATH"
+SCRIPT_PATH=$(pwd)
+
+# Only load library after changing to script directory.
+source lib.sh
+
+if [ -z "${LANGUAGES:-}" ]; then
+    # Load languages from config.
+    LANGUAGES=$(jq -r '(.sections_to_remove | keys | .[])' article_processing_config.json)
+fi
+# shellcheck disable=SC2086 # LANGUAGES is intentionally expanded.
+log "Selected languages:" $LANGUAGES
+
+TMP_DIR=$(mktemp --tmpdir -d wikiparser-download-XXXX)
+trap 'rm -rf $TMP_DIR' EXIT INT HUP
+
+log "Fetching run index"
+# Call wget outside of pipeline for errors to be caught by set -e.
+wget 'https://dumps.wikimedia.org/other/enterprise_html/runs/' --no-verbose  -O "$TMP_DIR/runs.html"
```

Comment:
Do you really need to store runs.html on disk and then clean it up?

Reply:
Good point, I had it like that for POSIX sh because there's no pipefail. With bash it shouldn't be a problem.